src/providers/timer-setting-storage/timer-setting-storage.ts

```diff
@@ -16,7 +16,10 @@ export class TimerSettingStorageProvider {
   constructor(public storage: Storage) {}
 
   public getTimerSettingByTimerId(timerId: number): Promise<TimerSetting> {
-    return this.storage.get(SETTING_STORAGE_KEY + timerId).then(setting => setting as TimerSetting);
+    return this.storage.get(SETTING_STORAGE_KEY + timerId).then(setting => {
+      if (!setting) throw Error('no Settings found')
+      return setting;
+    }).then(setting => setting as TimerSetting);
```

Comment:
couldn't that be done in the first then?
return setting as TimerSetting?

Reply:
Yes but this was mostly done to log future errors. This has since been heavily reworked. The first then is not even in it anymore as it was mostly to handle the time until I was able to delete the old timers without a setting and add a UI to manage timer settings.
Not sure if it's worth it to add it in here only to be overwritten later anyway...likely causing conflicts...